src/shared/components/ncTable/partials/TableCellProgress.vue

```diff
@@ -3,47 +3,141 @@
   - SPDX-License-Identifier: AGPL-3.0-or-later
 -->
 <template>
-	<div>
-		<NcProgressBar v-if="getValue !== null" :value="getValue" />
+	<div class="cell-progress" @dblclick="startEditing">
+		<div v-if="!isEditing">
+			<NcProgressBar v-if="getValue !== null" :value="getValue" />
+		</div>
+		<div v-else class="inline-editing-container">
+			<input
+				ref="input"
+				v-model.number="editValue"
+				type="number"
+				min="0"
+				max="100"
+				:disabled="localLoading"
+				class="cell-input"
+				@blur="saveChanges"
+				@keyup.enter="saveChanges"
+				@keyup.esc="cancelEdit">
+			<div v-if="localLoading" class="icon-loading-small icon-loading-inline" />
+		</div>
 	</div>
 </template>
 
 <script>
+import { showError } from '@nextcloud/dialogs'
+import { translate as t } from '@nextcloud/l10n'
+import { mapActions, mapState } from 'pinia'
+import { useDataStore } from '../../../../store/data.js'
 import { NcProgressBar } from '@nextcloud/vue'
 
 export default {
 	name: 'TableCellProgress',
-	components: { NcProgressBar },
+
+	components: {
+		NcProgressBar,
+	},
+
 	props: {
 		column: {
 			type: Object,
-			default: () => {},
+			required: true,
 		},
 		rowId: {
 			type: Number,
-			default: null,
+			required: true,
 		},
 		value: {
 			type: Number,
 			default: null,
 		},
 	},
+
+	data() {
+		return {
+			isEditing: false,
+			editValue: '',
+			localLoading: false,
+		}
+	},
+
 	computed: {
+		...mapState(useDataStore, {
+			rowMetadata(state) {
+				return state.getRowMetadata(this.rowId)
+			},
+		}),
+
 		getValue() {
 			if (this.value !== null && !isNaN(this.value)) {
 				return this.value
 			}
 			return null
 		},
 	},
+
+	methods: {
+		...mapActions(useDataStore, ['updateRow']),
+
+		startEditing() {
+			this.editValue = this.value
+			this.isEditing = true
+			this.$nextTick(() => {
+				this.$refs.input.focus()
+			})
+		},
+
+		async saveChanges() {
```

Comment:
This seems to be quite some duplication with the other components, maybe there is a reasonable way to make parts reusable here?